### DISCUSSION TITLE
Support for `setuptools_scm`?

### DISCUSSION THREAD

**Discussion by pmeier** [PyO3/maturin#1267] [Q&A] [answered]
I'm not sure how `maturin` builds the Python side of the project, so maybe this request makes no sense. If you are using `setuptools` would it be possible to add support for [`setuptools_scm`](https://github.com/pypa/setuptools_scm)? With that you don't have to manage the version manually. It is rather inferred from your SCM like git. Basically I want to use a dummy version in `Cargo.toml`.

If I use `build-backend = "setuptools.build_meta"` and have `setuptools_scm` is part of the build system requirements, it gets picked up automatically. However, using `build-backend = "maturin"` it is ignored.

**Comment by konstin** [comment]
For rust projects the workflows are different than with setuptools, i'd recommend checking out e.g. https://github.com/crate-ci/cargo-release

**Comment by messense** [comment, marked as answer]
`setuptools_scm` is designed for `setuptools` and [relies on extension APIs of `setuptools`](https://setuptools.pypa.io/en/latest/userguide/extension.html#customizing-distribution-options), maturin as a pep517 build backend doesn't even use `setuptools` so it can't support `setuptools_scm`.
